File: backend/app/models/concerns/status.rb
module Status
  module Enum
    extend ActiveSupport::Concern

    included do
      enum :status,
        {
          active: "active",
          inactive: "inactive",
          deleted: "deleted"
        },
        prefix: true,
        validate: true,
        default: :active

      default_scope -> { where.not(status: :deleted) }

      def destroy!
        update!(status: :deleted)
      end
    end
  end
end

File: frontend/src/app/constants/status.ts
import { BaseConstant } from "./baseconstant";

export class Status extends BaseConstant {
  protected static override _values = {
    active: { value: 1, label: "Ativo" },
    inactive: { value: 0, label: "Inativo" },
    deleted: { value: -1, label: "Excluído" }
  };

  static override options() {
    return [this._values.active, this._values.inactive];
  }
}
